src/rules/prefer-to-contain.ts

```diff
@@ -0,0 +1,248 @@
+ *
+ * @return {matcher is ParsedBooleanEqualityMatcher}
+ */
+const isBooleanEqualityMatcher = (
+  matcher: ParsedExpectMatcher,
+): matcher is ParsedBooleanEqualityMatcherCall =>
+  isParsedEqualityMatcherCall(matcher) &&
+  isBooleanLiteral(matcher.arguments[0]);
+
+type FixableIncludesCallExpression = KnownCallExpression<'includes'> &
+  CallExpressionWithSingleArgument;
+
+/**
+ * Checks if the given `node` is a `CallExpression` representing the calling
+ * of an `includes`-like method that can be 'fixed' (using `toContain`).
+ *
+ * @param {CallExpression} node
+ *
+ * @return {node is FixableIncludesCallExpression}
+ *
+ * @todo support `['includes']()` syntax (remove last property.type check to begin)
+ * @todo break out into `isMethodCall<Name extends string>(node: TSESTree.Node, method: Name)` util-fn
+ */
+const isFixableIncludesCallExpression = (
+  node: TSESTree.Node,
+): node is FixableIncludesCallExpression =>
+  node.type === AST_NODE_TYPES.CallExpression &&
+  node.callee.type === AST_NODE_TYPES.MemberExpression &&
+  isSupportedAccessor(node.callee.property, 'includes') &&
+  node.callee.property.type === AST_NODE_TYPES.Identifier &&
+  hasOnlyOneArgument(node);
+
+const buildToContainFuncExpectation = (negated: boolean) =>
+  negated ? `${ModifierName.not}.toContain` : 'toContain';
+
+/**
+ * Finds the first `.` character token between the `object` & `property` of the given `member` expression.
+ *
+ * @param {TSESTree.MemberExpression} member
+ * @param {SourceCode} sourceCode
+ *
+ * @return {Token | null}
+ */
+const findPropertyDotToken = (
+  member: TSESTree.MemberExpression,
+  sourceCode: TSESLint.SourceCode,
+) =>
+  sourceCode.getFirstTokenBetween(
+    member.object,
+    member.property,
+    token => token.value === '.',
+  );
+
+const getNegationFixes = (
+  node: FixableIncludesCallExpression,
+  modifier: NotNegatableParsedModifier,
+  matcher: ParsedBooleanEqualityMatcherCall,
+  sourceCode: TSESLint.SourceCode,
+  fixer: TSESLint.RuleFixer,
+  fileName: string,
+) => {
+  //.includes function argument
+  const [containArg] = node.arguments;
+  const negationPropertyDot = findPropertyDotToken(modifier.node, sourceCode);
+
+  const toContainFunc = buildToContainFuncExpectation(
+    matcher.arguments[0].value,
+  );
+  // ? 'not.toContain'
+  // : 'toContain';
+
+  /* istanbul ignore next */
+  if (negationPropertyDot === null) {
+    throw new Error(
+      `Unexpected null when attempting to fix ${fileName} - please file a github issue at https://github.com/jest-community/eslint-plugin-jest`,
+    );
+  }
+
+  return [
+    fixer.remove(negationPropertyDot),
+    fixer.remove(modifier.node.property),
+    fixer.replaceText(matcher.node.property, toContainFunc),
+    fixer.replaceText(matcher.arguments[0], sourceCode.getText(containArg)),
+  ];
+};
+
+const getCommonFixes = (
+  node: FixableIncludesCallExpression,
+  sourceCode: TSESLint.SourceCode,
+  fixer: TSESLint.RuleFixer,
```

Comment:
instead of passing in the fixer, could we just return the array and loop through them in the `fix` method itself?

Reply:
I'd prefer that too - however it'll make it harder to replace w/ `removeRange`.

Which depends on the answer to this question: do we care about persisting things like comments & spaces in fixers?

Currently, this will pass:

```
    {
      code: 'expect(/**/a/**/./**/includes/**/(/**/b/**/)/**/).toEqual(true);',
      errors: [{ messageId: 'useToContain', line: 1 }],
      output: 'expect(/**/a/**//**//**//**//**//**/).toContain(b);',
    },
    {
      code: 'expect(    a    .    includes    (    b    )    ).toEqual(true);',
      errors: [{ messageId: 'useToContain', line: 1 }],
      output: 'expect(    a                        ).toContain(b);',
    },
    {
      code:
        'expect(/**/a/**/[/**/"includes"/**/]/**/(/**/b/**/)/**/).toEqual(true);',
      errors: [{ messageId: 'useToContain', line: 1 }],
      output: 'expect(/**/a/**//**//**//**//**//**/).toContain(b);',
    },
    {
      code:
        'expect(    a    [    "includes"    ]    (    b    )    ).toEqual(true);',
      errors: [{ messageId: 'useToContain', line: 1 }],
      output: 'expect(    a                        ).toContain(b);',
    },
```

but if we use `removeRange`, it won't. In contrast however, the fixer becomes something like `removeRange(node.start, node.end)`, and makes it a lot easier for supporting square brackets.

Reply:
We should definitely preserver comments, otherwise it's not a safe fix. Whitespace is more meh, but preferably it should be kept